app/containers/GamePage/bang.js

```diff
@@ -0,0 +1,173 @@
+import React, { Component, PropTypes } from 'react';
+import { Button } from 'react-bootstrap';
+import Hand from './hand.js';
+
+export default class Bang extends Component {
+  static propTypes = {
+    assign: PropTypes.bool,
+    database: PropTypes.object,
+    email: PropTypes.string,
+    params: PropTypes.object,
+    participants: PropTypes.array,
+    players: PropTypes.number,
+    username: PropTypes.string,
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      hand: [],
+      roles: ['Sheriff', 'Outlaw', 'Renegade', 'Outlaw', 'Vice', 'Outlaw', 'Vice'],
+      deck: [],
+      joined: false,
+    };
+  }
+
+  assignRoles = () => {
+    const { roles } = this.state;
+    const { database, params, participants, players } = this.props;
+    const usersRef = database.ref(`${params.rid}/participants`);
+    const messagesRef = database.ref(`${params.rid}/messages`);
+    usersRef.off();
+    for (let i = players - 1; i >= 0; i--) {
+      const index = Math.floor(Math.random() * (i + 1));
+      const userRef = usersRef.child(participants[i].key);
+      participants[i].role = roles[index];
+      userRef.update({ role: participants[i].role });
+      userRef.update({ health: participants[i].health });
+      userRef.update({ hand: participants[i].hand });
+      if (roles[index] === 'Sheriff') {
+        messagesRef.push({ name: '[SYSTEM]', text: `${participants[i].name} is the Sheriff!` });
+      }
+      roles.splice(index, 1);
+      this.setState({ roles, participants });
+    }
+    this.createBangCards();
+    this.shuffleCards();
+    this.dealCards();
+    this.loadHand();
+  }
+
+  shuffleCards() {
+    const { deck } = this.state;
+    const { params, database } = this.props;
+    for (let i = deck.length - 1; i >= 0; i--) {
+      const rand = Math.floor(Math.random() * (i + 1));
```

Comment:
You sure this shouldnt be Math.random () * deck.length? instead of i + 1? Not sure how people do algorithm for shuffling cards tho but I think this logic youre only shuffling part of the deck at first arent you? So the first few cards are always the same whenever you start the game